dns.c

```diff
@@ -22,10 +22,12 @@ extern int res_search();
 static unsigned short getshort(c) unsigned char *c;
 { unsigned short u; u = c[0]; return (u << 8) + c[1]; }
 
-static union { HEADER hdr; unsigned char buf[PACKETSZ]; } response;
+static struct { unsigned char *buf; } response;
```

Comment:
What about something like

```c
struct resp {
  HEADER hdr;
  char *buf;
} *response;
response = malloc(256);
```

Then we would not need to cast things to HEADER later. And maybe we could just do it in one line so the case that this has size 0 doesn't ever happen.

Reply:
I haven't understood the original code well enough to try this. If you do, please push freely to #201.